src/rws_subscription.cpp

```diff
@@ -70,9 +78,12 @@ namespace abb :: rws
       }
       catch (Poco::TimeoutException const&)
       {
+        now = std::chrono::steady_clock::now();
         BOOST_THROW_EXCEPTION(
-          TimeoutError {"WebSocket frame receive timeout"}
+          TimeoutError {"WebSocket Failed to receive ping-pong message"}
             << boost::errinfo_nested_exception(boost::current_exception())
+            << PingPongTimeoutErrorInfo {ping_pong_timeout}
+            << LastFrameReceivedErrorInfo {std::chrono::duration_cast<std::chrono::microseconds>(now - last_frame)}
```

Comment:
If you get a `TimeourException`, the time must have reached `deadline`, so `std::chrono::steady_clock::now()` will be slightly more than `deadline`. What is the purpose of reporting this value?

Also, I don't understand how it works with the `Timespan`, could you explain?